### DISCUSSION TITLE
About the difference between the csv file I produced using  file "test_chgnet.py" and the csv file under the chgnet folder

### DISCUSSION THREAD

**Discussion by Claudia-Hello** [janosh/matbench-discovery#96] [Q&A] [answered]
Hi,
I tried to reproduce the predictions of the CHGNet model on the WBM test set and got the csv file as follows:
The three columns include"material_id"`, `"chgnet_energy"`, and `"chgnet_structure"`.

However, the contents of the `2023-12-21-chgnet-0.3.0-wbm-IS2RE.csv` file in the chgnet folder are as follows:
The three columns include `"material_id"`, `"chgnet_energy"`, and `"e_form_per_atom_chgnet"`.

As you can see, the last column of my predicted csv file is "chgnet_structure", while the last column of `2023-12-21-chgnet-0.3.0-wbm-IS2RE.csv` is `"e_form_per_atom_chgnet"`. Also, my predicted csv file results in only 5319, while the original `2023-12-21-chgnet-0.3.0-wbm-IS2RE.csv` file results in 256,963.

To summarize, I would like to ask, how did you get such predicted results as `2023-12-21-chgnet-0.3.0-wbm-IS2RE.csv` file? Can you tell me the details?
Thank you!

**Comment by janosh** [comment, marked as answer]
`test_chgnet.py` is designed to be run as a slurm job array where each sub task runs on a different split of the WBM test set.

https://github.com/janosh/matbench-discovery/blob/71a5edc666a50bcab757bd5211886610677ab24e/models/chgnet/test_chgnet.py#L35-L49

once all jobs are finished you run `join_chgnet_results.py` to collect all output files into 1 file and apply the MP formation energy corrections:

https://github.com/janosh/matbench-discovery/blob/71a5edc666a50bcab757bd5211886610677ab24e/models/chgnet/join_chgnet_results.py#L30-L50

`join_chgnet_results.py` saves the final output as `yyy-mm-dd-chgnet-0.3.0-wbm-IS2RE.csv.gz` to disk
